include/dlaf/matrix/matrix.h

```diff
@@ -68,18 +68,31 @@ class Matrix : public Matrix<const T, D> {
   ///
   /// @pre size.isValid(),
   /// @pre !blockSize.isEmpty().
-  Matrix(const LocalElementSize& size, const TileElementSize& block_size);
+  Matrix(const LocalElementSize& size, const TileElementSize& tile_size) noexcept
+      : Matrix<T, D>(Distribution(size, tile_size)) {}
 
-  /// Create a distributed matrix of size @p size and block size @p block_size on the given 2D
-  /// communicator grid @p comm.
+  /// Create a distributed matrix of size @p size block size @p tile_size and tile size @p tile_size
+  /// on the given 2D communicator grid @p comm.
   ///
   /// @pre size.isValid(),
   /// @pre !blockSize.isEmpty().
-  Matrix(const GlobalElementSize& size, const TileElementSize& block_size,
-         const comm::CommunicatorGrid& comm);
+  Matrix(const GlobalElementSize& size, const TileElementSize& tile_size,
+         const comm::CommunicatorGrid& comm) noexcept
+      : Matrix<T, D>(Distribution(size, tile_size, comm.size(), comm.rank(), {0, 0})) {}
 
   /// Create a matrix distributed according to the distribution @p distribution.
-  Matrix(Distribution distribution);
+  Matrix(Distribution distribution) noexcept : Matrix<const T, D>(std::move(distribution)) {
```

Comment:
Is this really `noexcept`? The memory allocation can of course fail.

Reply:
Removed `noexcept` from all the allocating constructors.